File: services/order/order-kafka/src/main/java/ru/jetmeal/order/kafka/OrderChangedMessage.java
package ru.jetmeal.order.kafka;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.Map;
import java.util.UUID;
import lombok.Builder;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
public record OrderChangedMessage(
        @JsonProperty("event_type") OrderChangedEventType eventType,
        @JsonProperty("order_id") UUID orderId,
        @JsonProperty("user_id") String userId,
        @JsonProperty("restaurant_id") String restaurantId,
        @JsonProperty("courier_id") UUID courierId,
        @JsonProperty("total_cost") BigDecimal totalCost,
        @JsonProperty("menu_items") Map<String, Integer> menuItems,
        @JsonProperty("comment") String comment,
        @JsonProperty("timestamp") Instant timestamp) {}

File: services/delivery/src/main/kotlin/ru/jetmeal/delivery/clients/order/OrderChangedMessage.kt
package ru.jetmeal.delivery.clients.order

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import ru.jetmeal.delivery.generated.model.Point
import java.math.BigDecimal
import java.time.Instant
import kotlin.uuid.Uuid

enum class OrderEventType {
	CREATED,
	UPDATED,
	CANCELLED,
	FINISHED,
}

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class OrderChangedMessage(
	@JsonProperty("event_type") val eventType: OrderEventType?,
	@JsonProperty("order_id") val orderId: Uuid?,
	@JsonProperty("user_id") val userId: String?,
	@JsonProperty("restaurant_id") val restaurantId: String?,
	@JsonProperty("location") val location: Point?,
	@JsonProperty("courier_id") val courierId: Uuid?,
	@JsonProperty("total_cost") val totalCost: BigDecimal?,
	@JsonProperty("menu_items") val menuItems: Map<String, Int>?,
	@JsonProperty("comment") val comment: String?,
	@JsonProperty("timestamp") val timestamp: Instant?
)
